Lib/pstats.py

```diff
@@ -470,14 +496,42 @@ def print_call_heading(self, name_size, column_title):
         if subheader:
             print(" "*name_size + "    ncalls  tottime  cumtime", file=self.stream)
 
-    def print_call_line(self, name_size, source, call_dict, arrow="->"):
+    def print_call_line(self, name_size, source, call_dict, arrow="->", sort_tuple=None, sort_type=None, sel_list=()):
         print(func_std_string(source).ljust(name_size) + arrow, end=' ', file=self.stream)
         if not call_dict:
             print(file=self.stream)
             return
-        clist = sorted(call_dict.keys())
+
+        if sort_tuple:
+            stats_list = []
+            calls_only = False
+            for func, value in call_dict.items():
+                if isinstance(value, tuple):
+                    nc, cc, tt, ct = value  # cProfile orders it this way
+                    # matches sort_arg_defs: cc, nc, tt, ct, module, line, name, stdname
+                    stats_list.append((cc, nc, tt, ct) + func +
+                                      (func_std_string(func),))
+                else:
+                    if not calls_only:
+                        if "time" in sort_type:
+                            raise TypeError("Caller/callee stats for %s do not have time information. "
+                                            "Try using cProfile instead of profile if you wish to record time by caller/callee."
+                                            % func_std_string(func))
+                        calls_only = True
+                    stats_list.append((None, value, None, None) + func +
+                                      (func_std_string(func),))
+
+            stats_list.sort(key=cmp_to_key(TupleComp(sort_tuple).compare))
+            funclist = [t[4:7] for t in stats_list]
+        else:
+            funclist = list(sorted(call_dict.keys()))
+
+        msg = ""
```

Comment:
Why is msg dynamic?

Reply:
This is the style of `eval_print_amount` which I didn't need to modify. A separate change could adjust that.